cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/ScheduleNotificationSubscriberService.java

```diff
@@ -206,7 +208,26 @@ protected void processNotification(ProgramScheduleStoreDataset scheduleStore,
         LOG.warn("Ignore notification that doesn't have a schedule {} associated with, {}", scheduleId, notification);
         return;
       }
-      jobQueue.addNotification(record, notification);
+      try {
+        jobQueue.addNotification(record, notification);
+      } catch (Exception e) {
+        emitScheduleJobNotificationFailureMetrics(record.getSchedule().getScheduleId().getApplication(),
+                                                  record.getSchedule().getScheduleId().getSchedule());
+        throw e;
+      }
+    }
+
+    private void emitScheduleJobNotificationFailureMetrics(String application, String schedule) {
+      if (metricsCollectionService == null) {
```

Comment:
nit: Can we add a `LOG.debug` here, `Unable to emit  Schedule Job Notification Failure Metrics`?

Reply:
done